cmd/cloud_schedule.go

```diff
@@ -142,20 +143,42 @@ func getCloudScheduleCmd(gs *state.GlobalState, client *cloudapi.K6CloudClient)
 	updateSchedule.Flags().BoolVar(&activate, "activate", false, "Activate the schedule")
 
 	// k6 cloud schedule delete
+	var testId int64
 	deleteSchedule := &cobra.Command{
 		Use:   "delete",
 		Short: "Delete a schedule",
 		Long:  "Delete a schedule",
-		Args:  cobra.MinimumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			scheduleId, err := strconv.ParseInt(args[0], 10, 64)
-			if err != nil {
-				return err
+
+			var scheduleId int64
+			var err error
+
+			if testId != 0 {
+				if len(args) > 0 {
+					return errors.New("Please do not pass a schedule_id argument when using the --test-id flag")
+				}
+				schedule, err := client.GetScheduleFromTestId(testId)
+				if err != nil {
+					return err
+				}
+				scheduleId = schedule.Id
+			} else {
+				if len(args) != 1 {
+					return errors.New("Usage: k6 cloud schedule delete SCHEDULE_ID")
+				}
+
+				scheduleId, err = strconv.ParseInt(args[0], 10, 64)
+				if err != nil {
+					return err
+				}
+
```

Comment:
not a fan that we have to go from cobra handled argument validation to manual implementation & duplication when porting to new commands

Reply:
I had to solve the same problem in `k6 cloud test run` https://github.com/fornfrey/k6-cloud-cli/blob/hackathon/cmd/cloud_test_.go#L548